src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerProjectLoader.cs

```diff
@@ -224,6 +218,25 @@ private async Task<bool> LoadOrReloadProjectAsync(ProjectToLoad projectToLoad, T
                 }
             }
 
+            using var newProjects = TemporaryArray<LoadedProject>.Empty;
+            foreach (var project in existingProjects)
+            {
+                if (projectsToRemove.Contains(project))
+                {
+                    project.Dispose();
+                }
+                else
+                {
+                    newProjects.Add(project);
+                }
```

Comment:
If we're going to re-create the array all the time, just do ImmutableArray.RemoveRange().